File: Android/TagMaster/src/depollsoft/tagmaster/barbershop/RemoteLocation.java
package depollsoft.tagmaster.barbershop;

import com.bindroid.trackable.TrackableField;

public class RemoteLocation {

  private TrackableField<String> uri = new TrackableField<String>();

  private TrackableField<String> type = new TrackableField<String>();

  public String getType() {
    return this.type.get();
  }

  public String getUri() {
    return this.uri.get();
  }

  public void setType(String value) {
    this.type.set(value);
  }

  public void setUri(String value) {
    this.uri.set(value);
  }
}

File: Android/TagMaster/src/depollsoft/tagmaster/barbershop/RemoteLocation.kt
package depollsoft.tagmaster.barbershop

import com.bindroid.trackable.*

class RemoteLocation {
    var uri by TrackableField<String>()
    var type by TrackableField<String>()

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is RemoteLocation) return false
        return uri == other.uri && type == other.type
    }

    override fun hashCode(): Int {
        var result = uri?.hashCode() ?: 0
        result = 31 * result + (type?.hashCode() ?: 0)
        return result
    }
}